### PR TITLE
feat: overhaul connector tests

### PR DESCRIPTION
**Pull request by joshgummersall**
- provide OAuth/JWT stub helpers
- expand JWT token validation tests
- reorg existing auth tests
- remove unnecessary `assertExt` helpers.

The bulk of this work is to protect me from inadvertent bugs caused by the incoming `CloudAdapter` port work. In particular, having a way to stub and test JWT/Oauth end-to-end with full signature verification is handy.

### CODE REVIEW

File: libraries/botbuilder-stdlib/src/assertExt.ts [hunk 1]
```diff
@@ -19,21 +18,3 @@ export function assertCondition(condition: unknown, message: string, ctor: Newab
...
     }
 }
-
-/**
- * Executes `block` and ensures it throws an exception with `message`
- *
- * @param {(...args: any[]) => any} block a function that should throw
- * @param {string} message expected error message
- */
-export function throwsMessage(block: (...args: unknown[]) => unknown, message: string): void {
```
Comment (joshgummersall) [inline libraries/botbuilder-stdlib/src/assertExt.ts:29]: Turns out I just didn't understand `assert.rejects` 😅

File: libraries/botframework-connector/src/auth/skillValidation.ts [hunk 2]
```diff
@@ -5,43 +5,44 @@
...
 
 /**
  * Validates JWT tokens sent to and from a Skill.
  */
-// eslint-disable-next-line @typescript-eslint/no-namespace
 export namespace SkillValidation {
-    /**
-     * TO SKILL FROM BOT and TO BOT FROM SKILL: Token validation parameters when connecting a bot to a skill.
-     */
-    const _tokenValidationParameters: VerifyOptions = {
```
Comment (joshgummersall) [inline libraries/botframework-connector/src/auth/skillValidation.ts:28]: Note: not exported and doesn't need to live inside the namespace.

File: libraries/botframework-connector/src/connectorApi/connectorClient.ts [hunk 3]
```diff
@@ -4,31 +4,31 @@
  * regenerated.
  */
 
-import * as msRest from "@azure/ms-rest-js";
```
Comment (joshgummersall) [inline libraries/botframework-connector/src/connectorApi/connectorClient.ts:7]: This file diff is purely code formatting.

File: libraries/botframework-connector/tests/auth.test.js [hunk 4]
```diff
@@ -1,434 +0,0 @@
-const assert = require('assert');
```
Comment (joshgummersall) [inline libraries/botframework-connector/tests/auth.test.js:1]: This was broken up into several discrete tests according to the class under test.

File: testing/botbuilder-test-utils/src/jwt.ts [hunk 5]
```diff
@@ -0,0 +1,127 @@
+// Copyright (c) Microsoft Corporation.
```
Comment (joshgummersall) [inline testing/botbuilder-test-utils/src/jwt.ts:1]: This magically allows us to fully stub JWT in the tests.

File: testing/botbuilder-test-utils/src/oauth.ts [hunk 6]
```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation.
```
Comment (joshgummersall) [inline testing/botbuilder-test-utils/src/oauth.ts:1]: And this the Oauth flow, including fully signed and verifiable JWTs (using the above stub).

**Review by stevengum — APPROVED** [review]
This looks very solid. Should this PR be crosslinked with https://github.com/microsoft/botframework-sdk/issues/5977 for the new botbuilder-test-utils `nock`-ing code?

**Pull request merged**